core/templates/dev/head/pages/topic-editor-page/modal-templates/select-skill-modal.template.html

```diff
@@ -1,31 +1,69 @@
+    <div ng-if="skillSummaries.length == 0">
+      You can only create questions for topics that have atleast one skill
+      attached to it. Please link a skill to this topic before adding questions.
+    </div>
+    <div class="list-group">
+      <a class="list-group-item protractor-test-skill-item"
+         ng-repeat="skillSummary in skillSummaries track by $index"
+         ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
+         ng-class="{'list-group-item-success': skillSummary.isSelected}">
+        <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
+      </a>
+    </div>
   </div>
-  <div class="list-group">
-    <a class="list-group-item protractor-test-skill-item"
-       ng-repeat="skillSummary in skillSummaries track by $index"
-       ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
-       ng-class="{'list-group-item-success': skillSummary.isSelected}">
-      <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
-    </a>
+  <div class="modal-footer" ng-switch="skillSummaries.length > 0">
+    <div ng-switch-when="true">
+      <button class="btn btn-default" ng-click="cancel()"> Cancel </button>
+      <button class="btn btn-success protractor-test-confirm-skill-button" ng-disabled="selectedSkillId === null" ng-click="next()"> Next </button>
+    </div>
+    <div ng-switch-when="false">
+      <button class="btn btn-default" ng-click="ok()"> Ok </button>
+    </div>
   </div>
 </div>
-<div class="modal-footer" ng-switch="skillSummaries.length > 0">
-  <div ng-switch-when="true">
-    <button class="btn btn-default" ng-click="cancel()"> Cancel </button>
-    <button class="btn btn-success protractor-test-confirm-skill-button" ng-disabled="selectedSkillId === null" ng-click="done()"> Done </button>
+<!-- Select Difficulty View -->
+<div ng-if="selectDifficulty">
+  <div class="modal-header" ng-if="selectDifficulty">
+    <h3 ng-if="selectedSkillIds.length == 0">
+      Select skills you would like to link to the question.
+    </h3>
+    <h3 ng-if="selectedSkillIds.length > 0">
+      Select the difficulty per skill:
+    </h3>
+  </div>
+  <div class="modal-body select-skill-modal">
+    <div ng-if="selectedSkillIds.length == 0">
+      You can only set difficulty after selecting skills linked to the question. Please go back and link skill(s) to this question.
+    </div>
+    <div class="list-group">
+      <div class="list-group-item"  ng-repeat="skillId in selectedSkillIds track by $index">
+        <label class="list-group-item-heading" style="font-size: 1em;"><[selectedSkillDescriptions[$index]]></label>
+        <md-radio-group ng-model="selectedSkillDifficulties[$index]">
+          <md-radio-button value="0.3">Easy</md-radio-button>
+          <md-radio-button value="0.6"> Medium </md-radio-button>
+          <md-radio-button value="0.9">Hard</md-radio-button>
```

Comment:
Remove the spaces for "Medium" or add spaces for the other ones.

Reply:
Done